### PR TITLE
SIDM-6761 / SIDM-7177 - Added suspend functionality

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
### JIRA link ###

https://tools.hmcts.net/jira/browse/SIDM-6761

### Change description ###

- Added ability to suspend user
- Added unit tests for UserSuspendController

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/views/suspend-user-successful.njk [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{% from "macros/breadcrumb.njk" import breadcrumb %}
+ {% from "macros/csrf.njk" import csrfProtection %}
```
Comment (KianKwa) [inline src/main/views/suspend-user-successful.njk:2]: Looks like there are some indentation issues in this njk file

File: src/main/views/un-suspend-user-successful.njk [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{% from "macros/breadcrumb.njk" import breadcrumb %}
+ {% from "macros/csrf.njk" import csrfProtection %}
```
Comment (KianKwa) [inline src/main/views/un-suspend-user-successful.njk:2]: identation issue again

File: src/main/views/un-suspend-user.njk [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{% from "macros/breadcrumb.njk" import breadcrumb %}
+ {% from "macros/csrf.njk" import csrfProtection %}
```
Comment (KianKwa) [inline src/main/views/un-suspend-user.njk:2]: Another indentation issue

File: src/main/views/suspend-user.njk [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{% from "macros/breadcrumb.njk" import breadcrumb %}
+ {% from "macros/csrf.njk" import csrfProtection %}
```
Comment (KianKwa) [inline src/main/views/suspend-user.njk:2]: indentation issue

File: src/main/views/user-details.njk [hunk 5]
```diff
@@ -141,7 +141,8 @@
     <input type="hidden" name="_userId" value="{{ content.user.id }}">
     <div class="govuk-button-group">
       {{ govukButton({ text: "Edit user", name: "_action", value: "edit", classes: "govuk-button" }) }}
-      {{ govukButton({ text: "Suspend user", name: "_action", value: "suspend", classes: "govuk-button--secondary" }) }}
+      {{ govukButton({ text: "Suspend user", name: "_action", value: "suspend", classes: "govuk-button--secondary" }) if content.user.active }}
+      {{ govukButton({ text: "Un-suspend user", name: "_action", value: "un-suspend", classes: "govuk-button--secondary" }) if not content.user.active }}
```
Comment (KianKwa) [inline src/main/views/user-details.njk:145]: The condition for showing the unsuspend button "if not content.user.active"....does it handle the case where the user is "Archived"? I presume we don't want to show the button is the user status is "Archived"

File: src/main/views/suspend-user.njk (hunk 4 above)
Comment (Thomas-Geraghty) [reply src/main/views/suspend-user.njk]: Resolved in https://github.com/hmcts/idam-user-dashboard/pull/238/commits/dcd4ffded02acb65dd55dc7a391e28ad8cb4770d

File: src/main/views/un-suspend-user.njk (hunk 3 above)
Comment (Thomas-Geraghty) [reply src/main/views/un-suspend-user.njk]: Resolved in https://github.com/hmcts/idam-user-dashboard/pull/238/commits/dcd4ffded02acb65dd55dc7a391e28ad8cb4770d

File: src/main/views/un-suspend-user-successful.njk (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/main/views/un-suspend-user-successful.njk]: Resolved in https://github.com/hmcts/idam-user-dashboard/pull/238/commits/dcd4ffded02acb65dd55dc7a391e28ad8cb4770d

File: src/main/views/suspend-user-successful.njk (hunk 1 above)
Comment (Thomas-Geraghty) [reply src/main/views/suspend-user-successful.njk]: Resolved in https://github.com/hmcts/idam-user-dashboard/pull/238/commits/dcd4ffded02acb65dd55dc7a391e28ad8cb4770d

File: src/main/views/user-details.njk (hunk 5 above)
Comment (Thomas-Geraghty) [reply src/main/views/user-details.njk:145]: All buttons now hidden when user stale in https://github.com/hmcts/idam-user-dashboard/pull/238/commits/5ff3131225f0041e109daed3e25a65304b8a7b34

**Review by KianKwa — APPROVED** [review]

**Pull request merged**